src/python/pants/engine/target.py

```diff
@@ -2636,6 +2636,20 @@ def any_are_covered_by_includes(self, addresses: Iterable[Address]) -> bool:
             for addr in addresses
         )
 
+    @memoized_method
+    def covered_by_includes(self, addresses: frozenset[Address]) -> frozenset[Address]:
+        """Return only the addresses which are in the explicitly provided includes."""
+        out = set()
+        for addr in addresses:
+            if addr in self.includes:
+                out.add(addr)
+            maybe_tgt_generator = addr.maybe_convert_to_target_generator()
+            if maybe_tgt_generator in self.includes:
+                # TODO: do we include the maybe_tgt_generator or the addr?
```

Comment:
I'm honestly not sure about this. I should also probably add a test to pin down the desired behaviour